internal/images/images.go

```diff
@@ -386,6 +390,78 @@ func decodeHEIFConfig(rat io.ReaderAt) (image.Config, error) {
 	}, nil
 }
 
+// parseLeadingNumber strips any unit suffix and converts the leading
+// number to float64.  Returns ok=false if it can’t.
+func parseLeadingNumber(s string) (v float64, ok bool) {
+	s = strings.TrimSpace(s)
+	i := 0
+
+	for _, r := range s {
+		if unicode.IsDigit(r) || r == '+' || r == '-' || r == '.' {
+			i++
+		} else {
+			break
+		}
+	}
+	if i == 0 {
+		return 0, false
+	}
+	n, err := strconv.ParseFloat(s[:i], 64)
+	return n, err == nil
+}
+
+// dimsFromSVG reads tokens until it sees the first <svg>
+// start-element, then extracts width/height/viewBox.
```

Comment:
the "viewBox" thing wasn't obvious to me (as somebody who doesn't know SVG) ... looked like a documentation mistake because the signature on the next line had width & height but no viewbox mentioned.

Maybe just say:

// dimsFromSVG reads XML tokens from SVG file r until it can compute
// the dimensions of the image (in "abstract user units").